tests/test_api.py

```diff
@@ -722,3 +729,41 @@ def test_create_build_config_auto_start(self):
 
         build_response = osbs._create_build_config_and_build(build_request)
         assert build_response.json == {'spam': 'maps'}
+
+    def test_scratch_build_config(self):
+        config = Configuration()
+        osbs = OSBS(config, config)
+
+        build_json = {
+            'apiVersion': osbs.os_conf.get_openshift_api_version(),
+
+            'metadata': {
+                'name': 'build',
+                'labels': {
+                    'git-repo-name': 'reponame',
+                    'git-branch': 'branch',
+                },
+            },
+        }
+
+        build_request = flexmock(
+            render=lambda: build_json,
+            is_auto_instantiated=lambda: False,
+            scratch=True)
+
+        updated_build_json = copy.deepcopy(build_json)
+        updated_build_json['kind'] = 'Build'
+        updated_build_json['metadata']['labels']['scratch'] = 'true'
+        updated_build_json['spec'] = {}
+        updated_build_json['spec']['serviceAccount'] = 'builder'
+        build_name = 'scratch-%s' % datetime.datetime.now().strftime('%Y%m%d%H%M%S')
+        updated_build_json['metadata']['name'] = build_name
+
+        (flexmock(osbs.os)
+            .should_receive('create_build')
+            .with_args(updated_build_json)
+            .once()
+            .and_return(flexmock(json=lambda: {'spam': 'maps'})))
```

Comment:
Could add a flexmock(osbs.os).should_receive('create_build_config').never() to make that explicit.